src/index.js

```diff
@@ -54,25 +55,30 @@ class Libp2p extends EventEmitter {
 
     this.datastore = this._options.datastore
     this.peerInfo = this._options.peerInfo
-    this.peerBook = this._options.peerBook || new PeerBook()
+    this.peerStore = new PeerStore()
 
     this._modules = this._options.modules
     this._config = this._options.config
     this._transport = [] // Transport instances/references
     this._discovery = [] // Discovery service instances/references
 
     // create the switch, and listen for errors
-    this._switch = new Switch(this.peerInfo, this.peerBook, this._options.switch)
+    this._switch = new Switch(this.peerInfo, this.peerStore, this._options.switch)
 
     // Setup the Upgrader
     this.upgrader = new Upgrader({
       localPeer: this.peerInfo.id,
       onConnection: (connection) => {
         const peerInfo = getPeerInfo(connection.remotePeer)
+
+        this.peerStore.put(peerInfo)
```

Comment:
The put here doesn't do anything. `connection.remotePeer` is a PeerId, so `getPeerInfo` is just going to check the peerStore anyway. I would delete the `getPeerInfo` usage and just do: 
```js
const peerInfo = this.peerStore.get(connection.remotePeer.toB58String())
```

Reply:
When a `peerB` dials `peerA` the `peerStore` for `peerA` does not have the `peerB` on it. That way, we would have `peerInfo` = `undefined`
`getPeerInfo` puts in the `peerStore` if it is provided. I changed for the following:

```js
const peerInfo = getPeerInfo(connection.remotePeer, this.peerStore)

this.registrar.onConnect(peerInfo, connection)
this.emit('peer:connect', peerInfo)
```

Reply:
Oh wait, if I do that `getPeerInfo` tries to `put` invalid `PeerInfo` to the store in some scenarios from other tests.